satellite/accounting/live/redis.go

```diff
@@ -0,0 +1,77 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package live
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/skyrings/skyring-common/tools/uuid"
+	"go.uber.org/zap"
+
+	"storj.io/storj/storage"
+	"storj.io/storj/storage/redis"
+)
+
+type redisLiveAccounting struct {
+	log *zap.Logger
+
+	client *redis.Client
+}
+
+func newRedisLiveAccounting(log *zap.Logger, address string) (Service, error) {
+	client, err := redis.NewClientFrom(address)
+	if err != nil {
+		return nil, err
+	}
+	return &redisLiveAccounting{
+		log:    log,
+		client: client,
+	}, nil
+}
+
+// GetProjectStorageUsage gets inline and remote storage totals for a given
+// project, back to the time of the last accounting tally.
+func (cache *redisLiveAccounting) GetProjectStorageUsage(ctx context.Context, projectID uuid.UUID) (inlineTotal, remoteTotal int64, err error) {
+	marshalled, err := cache.client.Get(ctx, []byte(projectID.String()))
+	if err != nil {
+		if storage.ErrKeyNotFound.Has(err) {
+			return 0, 0, nil
+		}
+		return 0, 0, err
+	}
+	var curVal spaceUsedAccounting
+	err = json.Unmarshal(marshalled, &curVal)
+	if err != nil {
+		return 0, 0, err
+	}
+	return curVal.InlineSpace, curVal.RemoteSpace, nil
+}
+
+// AddProjectStorageUsage lets the live accounting know that the given
+// project has just added InlineSpaceUsed bytes of inline space usage
+// and RemoteSpaceUsed bytes of remote space usage.
+func (cache *redisLiveAccounting) AddProjectStorageUsage(ctx context.Context, projectID uuid.UUID, InlineSpaceUsed, RemoteSpaceUsed int64) error {
+	curInlineTotal, curRemoteTotal, err := cache.GetProjectStorageUsage(ctx, projectID)
```

Comment:
Add tests for the concurrent calls.

For example, let's say you do `100` concurrent calls to `AddProjectStorageUsage` then the result should be the total.